### DISCUSSION TITLE
Correct usage of `StimulusEpoch`

### DISCUSSION THREAD

**Discussion by bjhardcastle** [AllenNeuralDynamics/aind-data-schema#838] [Q&A] [answered]
A `StimulusEpoch` has only one stimulus, but we have epochs that present multiple stimuli randomly interleaved (epochs with multiple shuffled opto waveforms, or visual + auditory stimuli interleaved).

I think the only way to do this at the moment is to have multiple `StimulusEpoch` entries with the same start/stop times - is that the intended usage?

**Comment by saskiad** [comment, marked as answer]
I'm working on a revamp of `StimulusEpoch` because it's got ... a bunch of issues. I think it should accomodate this - basically allowing multiple types of stimuli - I'll show it to you when it's ready to merge so we can make sure this will accomodate it.

**Reply by bjhardcastle** [reply]
Ok, then that's good timing. Thank you!
